packages/dialog/test/header-footer.test.js

```diff
@@ -414,4 +415,114 @@ describe('header/footer feature', () => {
       expect(getComputedStyle(headerPart).display).to.be.equal('none');
     });
   });
+
+  describe('overflow attribute', () => {
+    let content;
+
+    beforeEach(() => {
+      dialog.headerRenderer = createRenderer('Header');
+      dialog.footerRenderer = createRenderer('Footer');
+      dialog.renderer = createRenderer(Array(10).join('Lorem ipsum dolor sit amet\n'));
+      dialog.resizable = true;
+      dialog.opened = true;
+      content = overlay.$.content;
+      overlay.style.maxWidth = '300px';
+    });
+
+    describe('resize', () => {
+      const nextResize = (target) => {
+        return new Promise((resolve) => {
+          new ResizeObserver(() => setTimeout(resolve)).observe(target);
+        });
+      };
+
+      it('should not set overflow attribute when content has no scrollbar', () => {
+        expect(overlay.hasAttribute('overflow')).to.be.false;
+      });
+
+      it('should set overflow attribute when scrollbar appears on resize', async () => {
+        overlay.style.maxHeight = '320px';
+        await nextResize(overlay);
+
+        expect(overlay.getAttribute('overflow')).to.equal('bottom');
+      });
+
+      it('should remove overflow attribute when header renderer is removed', async () => {
+        overlay.style.maxHeight = '320px';
+        await nextResize(overlay);
+
+        dialog.headerRenderer = null;
+
+        expect(overlay.hasAttribute('overflow')).to.be.false;
+      });
+
+      it('should remove overflow attribute when footer renderer is removed', async () => {
+        overlay.style.maxHeight = '320px';
+        await nextResize(overlay);
+
+        dialog.footerRenderer = null;
+
+        expect(overlay.hasAttribute('overflow')).to.be.false;
+      });
+
+      it('should set overflow attribute when header title property is set', async () => {
+        dialog.headerRenderer = null;
+
+        overlay.style.maxHeight = '320px';
```

Comment:
nit: to make sure the `overflow` attribute works correctly with `headerTitle` alone (and not only because the footer has some content):

```suggestion
        dialog.headerRenderer = null;
        dialog.footerRenderer = null;

        overlay.style.maxHeight = '200px';
```